var/spack/repos/builtin/packages/pnmpi/package.py

```diff
@@ -0,0 +1,60 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+from spack import *
+
+
+class Pnmpi(CMakePackage):
+    """PnMPI is a dynamic MPI tool infrastructure that builds on top of
+       the standardized PMPI interface. """
+
+    homepage = "https://github.com/LLNL/PnMPI"
+    url      = "https://github.com/LLNL/PnMPI/releases/download/v1.7/PnMPI-v1.7-full.tar.gz"
+
+    version('1.7', '8040c1558c0deaa3d964c35d1760f3a8')
+
+    variant('fortran', default=True,
+            description='Configure PnMPI with Fortran support')
+    variant('tests', default=True,
+            description='Build test cases and enable "test" makefile target')
+
+    depends_on('cmake', type='build')
+    depends_on('argp-standalone', when='platform=darwin')
+    depends_on('binutils')
+    depends_on('help2man')
+    depends_on('doxygen')
+    depends_on('mpi')
+
+    def cmake_args(self):
+        args = []
+        spec = self.spec
+        on_off = {True: 'ON', False: 'OFF'}
+
+        has_fortran = spec.satisfies('+fortran')
```

Comment:
if you like, additional to the options below, you can test if a Fortran compiler exists:
```python
    @run_before('cmake')
    def die_without_fortran(self):
        if self.spec.satisfies('+fortran'):
            if (self.compiler.f77 is None) or (self.compiler.fc is None):
                raise InstallError(
                    'PnMPI Fortran variant requested but no Fortran compiler found!'
                )
```

(untested)

Reply:
This method is a good idea. I wonder if it's possible to move it up in the inheritance hierarchy?  There are many packages that depend on Fortran (optionally or otherwise), and if I remember correctly, few of them test for Fortran compilers before attempting to build the software.

Reply:
Added. I replaced tests like `self.compiler.f77 is None` tests with `not self.compiler.f77`, which is more idiomatic Python, and reduced the amount of nesting. I also haven't tested it otherwise, because I'm a little hesitant to alter my compiler setup.

Reply:
Thanks, that's great!

> This method is a good idea. I wonder if it's possible to move it up in the inheritance hierarchy? There are many packages that depend on Fortran (optionally or otherwise),

Currently not I am afraid, but we discussed in the past that compilers might be expressed just as virtual packages, "providing" C++98/11/14/17 or Fortran 77/90/20... functionality, at some point in the future. Maybe you want to ask on Slack or the mailing list about the progress?